tests/test_main_startup_rollback.py

```
@@ -0,0 +1,441 @@
+            "mmrelay.main.asyncio.get_running_loop",
+            side_effect=_make_patched_get_running_loop(),
+        ),
+        patch("mmrelay.main.asyncio.to_thread", side_effect=inline_to_thread),
+        patch("mmrelay.main.asyncio.Event", return_value=_ImmediateEvent()),
+        patch("mmrelay.main.meshtastic_utils.check_connection", new=_async_noop),
+        patch(
+            "mmrelay.main.meshtastic_utils.unsubscribe_meshtastic_callbacks"
+        ) as mock_unsubscribe,
+        patch("mmrelay.main.logger"),
+    ):
+        with pytest.raises(RuntimeError):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)